circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala

```diff
@@ -3,24 +3,81 @@ package org.virtuslab.ash.circe
 import java.io.NotSerializableException
 import java.nio.charset.StandardCharsets.UTF_8
 import java.util.NoSuchElementException
-
 import scala.jdk.CollectionConverters._
 import scala.reflect.ClassTag
 import scala.reflect.classTag
 import scala.reflect.runtime.{universe => ru}
-
 import akka.actor.ExtendedActorSystem
 import akka.event.Logging
 import akka.serialization.SerializerWithStringManifest
 import io.circe._
 import io.circe.jawn.JawnParser
 import org.reflections8.Reflections
 
+/**
+ * An abstract class that is extended to create a custom serializer.
+ *
+ * After creating your subclass, don't forget to add your serializer and base trait to `application.conf`
+ * (for more info [[https://doc.akka.io/docs/akka/2.5.32//serialization.html]])
+ *
+ * Example subclass:
+ * {{{
+ *   class CustomSerializer(actorSystem: ExtendedActorSystem) extends CirceAkkaSerializer[MySerializable](actorSystem) {
+ *
+ *     implicit private val serializabilityCodec: Codec[MySerializable] = genericCodec
+ *
+ *     override def identifier: Int = 41
+ *
+ *     override lazy val codecs =
+ *       Seq(
+ *         Register[SthCommand],
+ *         Register(implicitly[ru.TypeTag[ModifiedCodec]], prepareEncoder, prepareDecoder),
+ *         Register[GenericClass[MySerializable, MySerializable]])
+ *
+ *     override lazy val manifestMigrations = Seq("app.OldName" -> classOf[TopTraitMigration].getName)
+ *
+ *     override lazy val packagePrefix = "app"
+ *   }
+ * }}}
+ * @param system ExtendedActorSystem that is provided by Akka
+ * @tparam Ser base trait that is used to mark serialization
+ */
 abstract class CirceAkkaSerializer[Ser <: AnyRef: ClassTag](system: ExtendedActorSystem)
     extends SerializerWithStringManifest
     with AkkaCodecs {
+
+  /**
+   * Sequence that must contain [[org.virtuslab.ash.circe.Register#Registration]] for all direct subclasses of Ser.
+   *
+   * Each `Registration` is created using [[org.virtuslab.ash.circe.Register]]s [[org.virtuslab.ash.circe.Register#apply]] method.
+   *
+   * To check if all needed classes are registered, use Codec Registration Checker.
+   *
+   * @see [[org.virtuslab.ash.circe.Register]][[org.virtuslab.ash.circe.Register#apply]] for more information about type derivation
+   */
   val codecs: Seq[(ru.TypeTag[_ <: Ser], (Encoder[_ <: Ser], Decoder[_ <: Ser]))]
+
+  /**
+   * A sequence containing information used in type migration.
+   *
+   * If you ever change the name of a class, that is a direct descendant of `Ser`, and that class is persisted in any way, you must append new pair to this field.
+   * A pair of strings. The first one is the old FQCN and the second one is the new FQCN. The second `String` can be hardcoded, or better, extracted from `class`.
+   *
+   * Example:
+   * {{{
+   *   override lazy val manifestMigrations = Seq(
+   *    "app.OldName" -> "app.NewName",
```

Comment:
Btw maybe the second could be enforced to be a `Class` rather than a `String`? beyond the scope of this PR ofc

Reply:
Very hard to do even in compile time. This would require some kind of reference traversal until literal constant is found.

Reply:
Ok, nvm, I didn't understand comment fully xd

Reply:
Yeah, lets do it in #109

Reply:
Opened #110